common4j/src/testFixtures/java/com/microsoft/identity/http/HttpRequestRewriter.java

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.http;
+
+import com.microsoft.identity.common.java.net.HttpClient;
+import com.microsoft.identity.common.java.net.HttpResponse;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Map;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+
+/**
+ * An {@link HttpRequestInterceptor} that allows to re-write the http request and then proceeds
+ * with the modified request.
+ */
+@AllArgsConstructor
+public abstract class HttpRequestRewriter implements HttpRequestInterceptor {
```

Comment:
Should we add a new class for this or add these methods to the HttpRequestInterceptor itself..
I don't have strong preference, but feel that having a single place for all this kind of operation makes it more discoverable.

Reply:
We can certainly do either. Probably needs a more thorough analysis of pros and cons of each approach. In general, I think a lot of things can be improved with the interceptors. We've improved a few things here and we should definitely revisit this in the future to improve more things.